frontend/src/components/Toolbar.tsx

```diff
@@ -136,11 +141,14 @@ class Toolbar extends React.Component<ToolbarProps> {
             {/* Back Arrow */}
             {breadcrumbs.length > 0 &&
               <Tooltip title={'Back'} enterDelay={300}>
-                <IconButton className={css.backLink}
-                  // Need to handle this for when browsing back doesn't make sense
-                  onClick={this.props.history!.goBack}>
-                  <ArrowBackIcon className={css.backIcon} />
-                </IconButton>
+                <div> {/* Div needed because we sometimes disable a button within a tooltip */}
+                  <IconButton className={css.backLink}
+                    disabled={this.props.history!.length < 2}
```

Comment:
This check doesn't handle the case where the back button would take us outside of the Pipelines UI, but I don't think there's a way to handle that anyway. And maybe this behavior isn't very bad.